docs/content/sips/017-service-chaining.md

```diff
@@ -0,0 +1,106 @@
+title = "SIP 017 - Service Chaining"
+template = "main"
+date = "2024-02-21T23:00:00Z"
+
+---
+
+Summary: Allow HTTP requests to application components without going via the network
+
+Owner(s): ivan.towlson@fermyon.com
+
+Created: Feb 21, 2024
+
+# Requirements and scope
+
+In Spin 2.2, a component can make a HTTP request with only a path (e.g. `/api/accounts`).  Such a request will be routed to the same application, but this is done by Spin prepending the host where the application is running (e.g. `https://mysrsbizness.fermyon.app/api/accounts`).  The consequence of this is that requests go out to the network and come back in, including through load balancers etc.  This adds overhead, and in some environnments could count against quota or incur bandwidth charges.
+
+We have [prototyped](https://github.com/fermyon/spin/pull/2229) dispatching these requests directly to the component handling the route, but this was in the context of an implementation detail of Spin.  The requirement for this project is for an application to declare that it _requires_ dispatch to run locally on the node, and for hosts that cannot meet that prerequisite to reject the application.  This allows developers to write applications as a network of chained microservices, without having to be concerned about what happens if the microservices are not co-located.
+
+# User experience
+
+## Expressing a requirement for chaining
+
+To express that the application requires chaining for self-requests, add the following to `spin.toml`:
+
+```toml
+[application.prerequisites]
```

Comment:
Major nit: `prerequisites` threw me at first glance, how do you feel about `requirements` or `runtime_requirements`?

Reply:
A downside of "requirements": it is synonymous with "dependencies" in some places (Pythonlandia).

`features`? :grimacing:

Reply:
`[thingies]` it is!

Reply:
`[runtime_features]`?

Reply:
See also https://github.com/fermyon/spin/issues/2293 which may need some similar application-global runtime configuration (e.g. outbound http distributed trace headers).